engine/consensus/compliance/engine.go

```diff
@@ -63,6 +64,7 @@ type Engine struct {
 	finalizedView              counters.StrictMonotonousCounter
 	finalizationEventsNotifier engine.Notifier
 	con                        network.Conduit
+	detachedGoroutinesWg       sync.WaitGroup
```

Comment:
Hm, makes sense. I think that your fix works though. We are interested in situation where we have terminated component but goroutine still spawned because of race, but since the component has started termination process it means `ShutdownSignal` was closed and detached goroutine will be able to observe it and prematurely quit. 
I will apply your fix. Additionally I want to mention that this is a temporary fix. A proper fix for this behavior is in utilizing workers of `MessageHub` which is implemented in: https://github.com/onflow/flow-go/pull/3357